test/e2e/config/azure-dev.yaml

```diff
@@ -205,7 +205,7 @@ variables:
   # the management cluster to be upgraded.
   INIT_WITH_BINARY: "https://github.com/kubernetes-sigs/cluster-api/releases/download/v0.3.23/clusterctl-{OS}-{ARCH}"
   INIT_WITH_PROVIDERS_CONTRACT: "v1alpha3"
-  INIT_WITH_KUBERNETES_VERSION: "v1.21.2"
+  INIT_WITH_KUBERNETES_VERSION: "v1.21.14"
```

Comment:
seems like even with this change it's still using v1.21.2

```
I0628 20:17:17.935583       1 images.go:124] controller/azuremachine/virtualmachineimages.Service.getSKUAndVersion "msg"="Getting VM image SKU and version" "name"="clusterctl-upgrade-ddk37i-control-plane-dmzdt" "namespace"="clusterctl-upgrade-wzp1mw" "reconciler group"="infrastructure.cluster.x-k8s.io" "reconciler kind"="AzureMachine" "x-ms-correlation-request-id"="0d9e65d7-f0a2-4c8b-9f35-9a7ed0b10fa5" "k8sVersion"="v1.21.2" "location"="canadacentral" "offer"="capi" "osAndVersion"="ubuntu-2004" "publisher"="cncf-upstream"
```

Reply:
looks like that's because we're setting the variable in the test-infra job config https://github.com/kubernetes/test-infra/blob/master/config/jobs/kubernetes-sigs/cluster-api-provider-azure/cluster-api-provider-azure-presubmits-main.yaml#L453

PR incoming...